hazelcast/src/main/java/com/hazelcast/jet/pipeline/ServiceFactories.java

```diff
@@ -91,7 +98,9 @@ private ServiceFactories() { }
      */
     @Nonnull
     public static <K, V> ServiceFactory<?, IMap<K, V>> iMapService(@Nonnull String mapName) {
-        return ServiceFactories.sharedService(ctx -> ctx.hazelcastInstance().getMap(mapName));
+        ServiceFactory<?, IMap<K, V>> iMapServiceFactory =
+                ServiceFactories.sharedService(ctx -> ctx.hazelcastInstance().getMap(mapName));
+        return iMapServiceFactory.withPermission(new MapPermission(mapName, ACTION_CREATE, ACTION_READ));
```

Comment:
This code executes on the client. It's easy to remove the permission with reflection.